mongodb/src/mongodb/endpoint.bal

```diff
@@ -31,41 +31,40 @@ public type Client client object {
     handle datasource;
 
     # Gets called when the endpoint is being initialized during the package initialization.
-    public function __init(ClientEndpointConfig config) returns error? {
-        self.datasource = initClient(config);
+    public function __init(ClientEndpointConfig config) returns ClientError? {
+        self.datasource = check initClient(config);
     }
 
-    public remote function insert(string collectionName, json? queryString) returns error? {
+    public remote function insert(string collectionName, json? queryString) returns ServerError? {
```

Comment:
Why do `Client` object returns an `ServerError`? Shall we wrap it to a `ClientError` and return it?
For an example;
```
var result = insertData(self.datasource, java:fromString(collectionName), java:fromString(jsonString));
if (result is ServerError) {
    ClientError e = error(CLIENT_ERROR_REASON, message = "<message>", cause = result);
    return e;
}
```

Check the other places as well.

Reply:
ServerErrors were marked for exceptions thrown from the DB level itself and ClientError is thrown only when the config has any issues. Do you think any other better ways to differentiate both with another naming perhaps?

Reply:
I think for mongoDB it doesn't make sense to have Client/Server error separation. We don't have any server representation here. We are just clients connecting to an external mongoDB server.

Reply:
As discussed offline, renamed errors into ApplicationError and DatabaseError which is more meaningful in this case